### PR TITLE
Consolidating mdspan cython defitions

### PR DESCRIPTION
**Pull request by cjnolet**
We recently had two PRs merged (one in 22.12 and another in 23.02) that independently added cython defitions around the mdspan factory functions. 

I've consolidated them in this PR and also exposed the remaining needed bits for specifying layouts in cython as well.

### CODE REVIEW

File: python/pylibraft/pylibraft/cpp/kmeans.pxd [hunk 1]
```diff
@@ -33,6 +33,9 @@ from pylibraft.cpp.optional cimport optional
 cdef extern from "raft_distance/kmeans.hpp" \
         namespace "raft::cluster::kmeans::runtime":
 
+    ctypedef const float const_float
+    ctypedef const double const_double
```
Comment (benfred) [inline python/pylibraft/pylibraft/cpp/kmeans.pxd:37]: do we need these typedefs here?   It doesn't seem like they are being used

Comment (cjnolet) [reply python/pylibraft/pylibraft/cpp/kmeans.pxd:37]: Nope, those need to be removed

**Review by benfred — APPROVED** [review]
looks great!

**Comment by cjnolet** [comment]
I've added these changes to 1055 already. Closing.

**Pull request closed without merging**